File: dongyimai_manager_web/src/main/webapp/js/service/typeTemplateService.js
//服务层
app.service('typeTemplateService',function($http){
	    	
	//findAll
	this.findAll=function(){
		return $http.get('../typeTemplate/findAll.do');		
	}

	//findPage
	this.findPage=function(page,rows){
		return $http.get('../typeTemplate/findPage.do?page='+page+'&rows='+rows);
	}

	//findOne
	this.findOne=function(id){
		return $http.get('../typeTemplate/findOne.do?id='+id);
	}

	//add
	this.add=function(entity){
		return  $http.post('../typeTemplate/add.do',entity );
	}

	//update
	this.update=function(entity){
		return  $http.post('../typeTemplate/update.do',entity );
	}

	//dele
	this.dele=function(ids){
		return $http.get('../typeTemplate/delete.do?ids='+ids);
	}

	//条件、模糊查询+分页
	this.search=function(page,rows,searchEntity){
		return $http.post('../typeTemplate/search.do?page='+page+"&rows="+rows, searchEntity);
	}

});

File: dongyimai_sellergoods_interface/src/main/java/com/offcn/sellergoods/service/TypeTemplateService.java
package com.offcn.sellergoods.service;

import com.offcn.entity.PageResult;
import com.offcn.pojo.TbTypeTemplate;

import java.util.List;
import java.util.Map;

public interface TypeTemplateService {

	//findAll
	public List<TbTypeTemplate> findAll();
	
	//findPage
	public PageResult findPage(int pageNum, int pageSize);
	
	//add
	public void add(TbTypeTemplate type_template);
	
	//update
	public void update(TbTypeTemplate type_template);
	
	//findOne
	public TbTypeTemplate findOne(Long id);
	
	//delete
	public void delete(Long[] ids);

	//条件、模糊查询+分页
	public PageResult findPage(TbTypeTemplate type_template, int pageNum, int pageSize);

	/**
	 * 根据模板ID查询规格及规格选项列表
	 * @param typeId
	 * @return
	 */
	public List<Map>  findSpecList(Long typeId);

}